<!-- Launch script for testing the pre-integration module
    John P Morrison
    MIT/WHOI Joint Program
    15 APR 2025
    - Designed to be run on an existing bag file with acomms, DVL and IMU data
-->

<launch>
    <!-- Args that we would change at agent-to-agent level-->
    <arg name="agent_name" default="actor_0" doc="format: actor_0 or beacon_0"/>
    <arg name="source_addr" default="0" doc="int"/>
    <arg name="num_agents" default="1" />
    <arg name="num_slots" default="2" doc="number of TDMA slots, must be even"/>
    <arg name="active_slots" default="0,1" doc="comma-separated list of active TDMA slots"/>

    <!-- Set the ROSCONSOLE_FORMAT -->
    <env name="ROSCONSOLE_FORMAT0" value="[${severity}] [${time}] [${node}]: ${message}"/>

    <!-- Start the agent node -->
    <group ns="$(arg agent_name)">

        <!-- Shared Parameters -->
        <param name="agent_name" value="$(arg agent_name)" />
        <param name="modem_address" value="$(arg source_addr)" />
        <!-- <param name="source_addr" value="0" /> -->
        <param name="num_agents" value="$(arg num_agents)" />
        <param name="num_landmarks" value="2" />
        <!-- <param name="landmarks" value="{L0:[-71.7845,-39.6078,1.5], L1:[65.0832,25.6598,1.5]}" /> -->
        <!-- <param name="num_slots" value="$(eval 2 * num_agents)" /> -->
        <param name="sigma_range" value="1" />
        <param name="sound_speed" value="1486" />

        <!-- Launch communication and graph managers-->
        <node name="preintegration_test" pkg="spurdog_acomms" type="test_preintegration.py" output="screen" />

        <!-- Launch the odometry node -->
        <node name="odometry" pkg="spurdog_acomms" type="odometry" output="screen">
            <param name="velocity_source" value="dvl" />
            <!-- <param name="accel_noise_sigma" value="0.05"/> -->
            <!-- <param name="gyro_noise_sigma" value="0.001"/> -->
            <!-- <param name="accel_bias_rw_sigma" value="0.01"/> -->
            <!-- <param name="gyro_bias_rw_sigma" value="0.01"/> -->
            <!-- <param name="accel_bias_x" value="0.00065798"/>
            <param name="accel_bias_y" value="-0.0044168"/>
            <param name="accel_bias_z" value="-0.01147435"/> -->
            <!-- <param name="gyro_bias_x" value="-0.002113"/>
            <param name="gyro_bias_y" value="-0.000075"/>
            <param name="gyro_bias_z" value="0.000314"/> -->
            <!-- <param name="gyro_bias_x" value="-0.005"/>
            <param name="gyro_bias_y" value="0.0"/> -->
            <!-- <param name="gyro_bias_z" value="-0.0025"/> -->
        </node>

    </group>
</launch>